xmtp_cryptography/src/signature.rs

```diff
@@ -111,14 +112,58 @@ pub fn h160addr_to_string(bytes: H160) -> String {
     s
 }
 
+/// Check if an string is a valid ethereum address (valid hex and length 20).
+pub fn is_valid_ethereum_address<S: AsRef<str>>(address: S) -> bool {
+    let address = address.as_ref();
+    let address = address.strip_prefix("0x").unwrap_or(address);
+
+    if address.len() != 40 {
+        return false;
+    }
+
+    address.chars().all(|c| c.is_ascii_hexdigit())
+}
+
+/// Check if an ed25519 public signature key is valid.
+pub fn is_valid_ed25519_public_key<Bytes: AsRef<[u8]>>(public_key: Bytes) -> bool {
```

Comment:
Not sure if completely necessary; however we had a test checking for invalid member registration and I didn't see how this would be a negative, since it would leave less errors for intent publishing

Reply:
It's a nice touch, prevents confusion/misuse between removing installations/accounts